HW02/Ivan_Papushoi/ivan_papushoi_hw_02_t_02.rb

```diff
@@ -0,0 +1,38 @@
+# Method should return an array of formatted strings
+# containing information about post requests
+# lines that do not match the format should be ignored
+# if there is no suitable string at the input,
+# the method should return an empty array
+
+def receive_filelines
+  File.readlines(FILE_NAME)
+end
+
+def receive_post(file_lines)
+  file_lines.select { |line| line.downcase.include?('post') }
+end
+
+# Format "IP - - [date] "Method adress protocol" answer_code ***"
+
+def parse_posts_to_columns(string_array)
+  array_classified_strings = []
+  string_array.each do |string|
+    ip_match = string.match(/^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}/)[0]
+    date_match = string.match(/\[(.+)\]/)[0].delete '[]'
+    message_match = string.match(/\"(.+)\"/)[0].delete '" POST HTTP1.1'.upcase
+    parsed_string = { adr: ip_match, date: date_match, msg: message_match }
+    array_classified_strings << parsed_string
+  end
+  array_classified_strings
+end
+
+def format_to_output(input_array)
+  input_array.map { |str| "#{str[:date]} FROM: #{str[:adr]} TO: #{str[:msg]}" }
+end
+
+def output_info_post_request_formatted
+  format_to_output(parse_posts_to_columns(receive_post(receive_filelines)))
```

Comment:
Your code should tell a story ( something like this )
```suggestion
   split_file_by_lines
   select_valid_lines
   transform_lines
   show_result
```

Reply:
Rewrote my code, now it's something as you suggest